tiledb/cf/engines/netcdf4_engine.py

```diff
@@ -638,27 +638,35 @@ def _from_group_to_attr_per_array(
                 from. Use ``'/'`` to specify the root group.
         """
         converter = cls(default_input_file, default_group_path)
+        coord_names = []
+        tiles_by_var = {} if tiles_by_var is None else tiles_by_var
+        tiles_by_dims = {} if tiles_by_dims is None else tiles_by_dims
         if coords_to_dims:
-            raise NotImplementedError
+            for ncvar in netcdf_group.variables.values():
+                if ncvar.ndim == 1 and ncvar.dimensions[0] == ncvar.name:
+                    converter.add_coord_to_dim_converter(ncvar)
+                    coord_names.append(ncvar.name)
         for ncvar in netcdf_group.variables.values():
-            for dim in ncvar.get_dims():
-                if dim.name not in converter.dim_names:
-                    converter.add_dim_to_dim_converter(
-                        dim,
-                        unlimited_dim_size,
-                        dim_dtype,
-                    )
+            if ncvar.name in coord_names:
+                continue
             if not ncvar.dimensions:
                 array_name = (
                     "scalars" if "scalars" not in netcdf_group.variables else "_scalars"
```

Comment:
This looks ambiguous; if a "scalars" array is added, it may or may not be in the original netcdf. Why not use the underscored name always? Or perhaps this doesn't matter?